### PR TITLE
Serverless Agent supports `stream-logs`

### PR DESCRIPTION
**Pull request by keisku**
### What does this PR do?

If `DD_SERVERLESS_STREAM_LOGS=true` is set, datadog-lambda-extension(serverless-agent) provides the feature like `agent stream-logs`.

### Motivation

- SLES-1598
- Better troubleshooting for log collection on the Lambda.

### Additional Notes

Used the implementation of `agent stream-logs` as a reference.

https://github.com/DataDog/datadog-agent/blob/d0b647bcb646076ba11398a169e28eabcea58a3e/comp/api/api/utils/stream/stream.go#L27-L94

#### Datadog Logs Explorer

There are not stream-logs output having the prefix, `DD_EXTENSION | stream-logs |`.

#### CloudWatch Logs

We can confirm what logs are sent to Datadog from the stream-logs output.

#### Dockerfile

Used this Dockerfile to deploy serverless agent to the testing Lambda env.

```dockerfile
FROM golang:1.22 as app-builder
# ...skip...

FROM golang:1.22 as datadog-lambda-extension-builder
WORKDIR /build
RUN git clone --branch keisku/serverless-stream-logs --depth=1 https://github.com/DataDog/datadog-agent.git
WORKDIR /build/datadog-agent
RUN go build -ldflags="-s -w" -o datadog-agent -tags serverless,otlp ./cmd/serverless

FROM public.ecr.aws/lambda/provided:al2023
RUN dnf update -y \
    && dnf install -y ca-certificates
COPY --from=app-builder /build/app /app
COPY --from=datadog-lambda-extension-builder /build/datadog-agent/datadog-agent /opt/extensions/datadog-agent
ENTRYPOINT [ "/app" ]
```

### CODE REVIEW

File: pkg/serverless/streamlogs/streamlogs_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+				},
+				{
+					message: newTestMessage("ログ2"),
+					rendered: []byte("ログ2"),
+				},
+				{
+					// When logs-agent stops, nil message is sent to stream-logs
+					message: nil,
+					rendered: nil,
+				},
```
Comment (keisku) [inline pkg/serverless/streamlogs/streamlogs_test.go:93]: To make sure this panic will not happen.

File: pkg/serverless/streamlogs/streamlogs_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+				{
+					message:  newTestMessage("ログ2"),
+					rendered: []byte("ログ2"),
+				},
+				{
+					// When logs-agent stops, nil message is sent to stream-logs
+					message:  nil,
+					rendered: nil,
+				},
```
Comment (keisku) [inline pkg/serverless/streamlogs/streamlogs_test.go:93]: To make sure this panic will not happen.

**Review by kayayarai — APPROVED** [review]
👍 from docs

File: pkg/logs/diagnostic/format.go [hunk 3]
```diff
@@ -32,18 +32,11 @@ func (l *logFormatter) Format(m *message.Message, eventType string, redactedMsg
...
-	if !m.ServerlessExtra.Timestamp.IsZero() {
-		ts = m.ServerlessExtra.Timestamp
-	}
-
 	return fmt.Sprintf("Integration Name: %s | Type: %s | Status: %s | Timestamp: %s | Hostname: %s | Service: %s | Source: %s | Tags: %s | Message: %s\n",
 		m.Origin.LogSource.Name,
 		m.Origin.LogSource.Config.Type,
 		m.GetStatus(),
-		ts,
+		time.Now().UTC(),
```
Comment (purple4reina) [inline pkg/logs/diagnostic/format.go:39]: Will this change the timestamp on serverless log messages? Help me understand this update.

Comment (purple4reina) [reply pkg/logs/diagnostic/format.go:39]: Okay, I think I understand now. Serverless stream logs will no longer be processed using this method, so you are removing the serverless specific code.

Comment (keisku) [reply pkg/logs/diagnostic/format.go:39]: @purple4reina Yes, the serverless-agent will depend on `streamlogs.Formatter`. We no longer need to consider `diagnostic.logFormatter` being called by the serverless-agent.

**Review by gh123man — APPROVED** [review]
AML owned files LGTM

**Review by purple4reina — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
@DataDog/agent-processing-and-routing files look good 👍

**Review by ogaca-dd — APPROVED** [review]
LGTM for files owned by ASC

**Review by purple4reina — APPROVED** [review]
I published a version of the serverless extension using this branch to test this out in a real live lambda function. I made a python lambda function that prints "Hello, world!". I am seeing the following log output from my function after two executions:

```
INIT_START Runtime Version: python:3.9.v53	Runtime Version ARN: arn:aws:lambda:sa-east-1::runtime:61e546f90702da13effe2459af74f33c6be93e79921e46f165413c75ec9bacba
TELEMETRY	Name: datadog-agent	State: Subscribed	Types: [Platform, Function, Extension]
EXTENSION	Name: datadog-agent	State: Ready	Events: [INVOKE, SHUTDOWN]
START
Hello, world!
DD_EXTENSION | stream-logs | Integration Name: AWS Logs | Type: string_channel | Status: info | Timestamp: 2024-06-04 15:45:37.895 +0000 UTC | Service: agent | Source: lambda | Tags: git.commit.sha:cdc39c43009185b747f6c9e2940122af28e6d1e7,architecture:x86_64,dd_extension_version:135,aws_account:425362996713,function_arn:arn:aws:lambda:sa-east-1:425362996713:function:rey-python-lambda-dev-simple,account_id:425362996713,runtime:python3.9,resource:rey-python-lambda-dev-simple,memorysize:1024,git.repository_url:github.com/purple4reina/dd-sample-apps.git,region:sa-east-1,functionname:rey-python-lambda-dev-simple | Message: START RequestId: 4e822b4a-037b-4a60-aeac-aba90aa85b3b Version: $LATEST
DD_EXTENSION | stream-logs | Integration Name: AWS Logs | Type: string_channel | Status: info | Timestamp: 2024-06-04 15:45:37.896 +0000 UTC | Service: agent | Source: lambda | Tags: git.commit.sha:cdc39c43009185b747f6c9e2940122af28e6d1e7,architecture:x86_64,dd_extension_version:135,aws_account:425362996713,function_arn:arn:aws:lambda:sa-east-1:425362996713:function:rey-python-lambda-dev-simple,account_id:425362996713,runtime:python3.9,resource:rey-python-lambda-dev-simple,memorysize:1024,git.repository_url:github.com/purple4reina/dd-sample-apps.git,region:sa-east-1,functionname:rey-python-lambda-dev-simple | Message: Hello, world!
DD_EXTENSION | stream-logs | Integration Name: AWS Logs | Type: string_channel | Status: info | Timestamp: 2024-06-04 15:45:37.907 +0000 UTC | Service: agent | Source: lambda | Tags: git.commit.sha:cdc39c43009185b747f6c9e2940122af28e6d1e7,architecture:x86_64,dd_extension_version:135,aws_account:425362996713,function_arn:arn:aws:lambda:sa-east-1:425362996713:function:rey-python-lambda-dev-simple,account_id:425362996713,runtime:python3.9,resource:rey-python-lambda-dev-simple,memorysize:1024,git.repository_url:github.com/purple4reina/dd-sample-apps.git,region:sa-east-1,functionname:rey-python-lambda-dev-simple | Message: END RequestId: 4e822b4a-037b-4a60-aeac-aba90aa85b3b
END Duration: 591.36 ms (init: 719.16 ms) Memory Used: 105 MB
START
Hello, world!
DD_EXTENSION | stream-logs | Integration Name: AWS Logs | Type: string_channel | Status: info | Timestamp: 2024-06-04 15:45:38.486 +0000 UTC | Service: agent | Source: lambda | Tags: git.commit.sha:cdc39c43009185b747f6c9e2940122af28e6d1e7,architecture:x86_64,dd_extension_version:135,aws_account:425362996713,function_arn:arn:aws:lambda:sa-east-1:425362996713:function:rey-python-lambda-dev-simple,account_id:425362996713,runtime:python3.9,resource:rey-python-lambda-dev-simple,memorysize:1024,git.repository_url:github.com/purple4reina/dd-sample-apps.git,region:sa-east-1,functionname:rey-python-lambda-dev-simple | Message: REPORT RequestId: 4e822b4a-037b-4a60-aeac-aba90aa85b3b	Duration: 591.36 ms	Runtime Duration: 12.00 ms	Post Runtime Duration: 579.36 ms	Billed Duration: 592 ms	Memory Size: 1024 MB	Max Memory Used: 105 MB	Init Duration: 719.16 ms
DD_EXTENSION | stream-logs | Integration Name: AWS Logs | Type: string_channel | Status: info | Timestamp: 2024-06-04 15:48:22.267 +0000 UTC | Service: agent | Source: lambda | Tags: git.commit.sha:cdc39c43009185b747f6c9e2940122af28e6d1e7,architecture:x86_64,dd_extension_version:135,aws_account:425362996713,function_arn:arn:aws:lambda:sa-east-1:425362996713:function:rey-python-lambda-dev-simple,account_id:425362996713,runtime:python3.9,resource:rey-python-lambda-dev-simple,memorysize:1024,git.repository_url:github.com/purple4reina/dd-sample-apps.git,region:sa-east-1,functionname:rey-python-lambda-dev-simple | Message: START RequestId: d4911f66-18e6-46a3-903f-bd8d9af196af Version: $LATEST
DD_EXTENSION | stream-logs | Integration Name: AWS Logs | Type: string_channel | Status: info | Timestamp: 2024-06-04 15:48:22.269 +0000 UTC | Service: agent | Source: lambda | Tags: git.commit.sha:cdc39c43009185b747f6c9e2940122af28e6d1e7,architecture:x86_64,dd_extension_version:135,aws_account:425362996713,function_arn:arn:aws:lambda:sa-east-1:425362996713:function:rey-python-lambda-dev-simple,account_id:425362996713,runtime:python3.9,resource:rey-python-lambda-dev-simple,memorysize:1024,git.repository_url:github.com/purple4reina/dd-sample-apps.git,region:sa-east-1,functionname:rey-python-lambda-dev-simple | Message: Hello, world!
DD_EXTENSION | stream-logs | Integration Name: AWS Logs | Type: string_channel | Status: info | Timestamp: 2024-06-04 15:48:22.27 +0000 UTC | Service: agent | Source: lambda | Tags: git.commit.sha:cdc39c43009185b747f6c9e2940122af28e6d1e7,architecture:x86_64,dd_extension_version:135,aws_account:425362996713,function_arn:arn:aws:lambda:sa-east-1:425362996713:function:rey-python-lambda-dev-simple,account_id:425362996713,runtime:python3.9,resource:rey-python-lambda-dev-simple,memorysize:1024,git.repository_url:github.com/purple4reina/dd-sample-apps.git,region:sa-east-1,functionname:rey-python-lambda-dev-simple | Message: END RequestId: d4911f66-18e6-46a3-903f-bd8d9af196af
END Duration: 1048.60 ms Memory Used: 106 MB
DD_EXTENSION | stream-logs | Integration Name: AWS Logs | Type: string_channel | Status: info | Timestamp: 2024-06-04 15:48:23.316 +0000 UTC | Service: agent | Source: lambda | Tags: git.commit.sha:cdc39c43009185b747f6c9e2940122af28e6d1e7,architecture:x86_64,dd_extension_version:135,aws_account:425362996713,function_arn:arn:aws:lambda:sa-east-1:425362996713:function:rey-python-lambda-dev-simple,account_id:425362996713,runtime:python3.9,resource:rey-python-lambda-dev-simple,memorysize:1024,git.repository_url:github.com/purple4reina/dd-sample-apps.git,region:sa-east-1,functionname:rey-python-lambda-dev-simple | Message: REPORT RequestId: d4911f66-18e6-46a3-903f-bd8d9af196af	Duration: 1048.60 ms	Runtime Duration: 3.00 ms	Post Runtime Duration: 1045.60 ms	Billed Duration: 1049 ms	Memory Size: 1024 MB	Max Memory Used: 106 MB
```

I was worried that printing each log as it comes in would create an infinite loop, but it does not.

**Pull request merged**